client/react-dapp/src/App.tsx

```diff
@@ -34,12 +34,26 @@ type Role = {
 };
 
 function App() {
-  const [roles, setRoles] = useState<Role[]>([]);
-  const [did, setDID] = useState<string>("");
+  const userRoles = localStorage.getItem("roles");
```

Comment:
What's the point of keeping the roles in local storage they are being refreshed in `useEffect`? (Assuming I'm understanding the point of `useEffect` correctly)

Reply:
The point of keeping the roles in the local storage is to persist the state which isn't available after a page refresh because the component resets to the initial state after that hence we won't be able to keep the user logged in.

The roles are not being refreshed in useEffect, useEffect  is basically the `componentDidMount`, `componentDidUpdate`, and `componentWillUnmount` lifecycle methods combined.  `componentDidMount` will get called after the page refresh so we are just going to fetch the roles and if they are returned, it means the user is still logged in and we don't do anything because the roles will be gotten from the localStorage after the component is mounted else we logged the user out for them to login again.

Reply:
Ah yes, I forgot that the `/roles` endpoint doesn't get any new roles that the user has, it only returns what is in the backend token already. For some reason I thought that it got the new roles from the user's DID document services. 

If the point of querying the `/roles` endpoint in `useEffect` is to check the log-in status of the user, wouldn't it be much clearer to use the new `login-status` endpoints that you have made?